lib/ansible/playbook/role/__init__.py

```diff
@@ -281,6 +321,31 @@ def _load_dependencies(self):
 
         return deps
 
+    def _create_arg_spec_validation_task_data(self, argument_spec, argument_spec_name,
+                                              role_name, role_path, role_params):
+        # If the arg spec provides a description, use it to flesh out the validation task name
+        task_name_parts = ["Validating arguments against arg spec '%s'" % argument_spec_name]
+        if 'description' in argument_spec:
+            task_name_parts.append(argument_spec['description'])
+        task_name = ' - '.join(task_name_parts)
+
+        arg_spec_task = {'action': {'module': 'validate_arg_spec',
+                                    'argument_spec': argument_spec,
+                                    'provided_arguments': role_params,
```

Comment:
role_params are not the only things valid, any variable provided is, just because a required variable is not provided as 'role_param' does not mean the role wont work. i.e
```
roles:
   - name: myrole
     thisisroleparam: othervalue
     vars:
        notroleparambutvalid: value
```
this should work with BOTH variables being required

Reply:
That works currently, added https://github.com/ansible/ansible/pull/44983/files#diff-c8709e49ceb22733ac4acc130e841708 as an test and example.